jetstream/stream_config.go

```diff
@@ -26,154 +26,356 @@ import (
+		// Lag is the number of uncommitted operations this peer is behind the
+		// leader.
+		Lag uint64 `json:"lag,omitempty"`
 	}
 
-	// SubjectTransformConfig is for applying a subject transform (to matching messages) before doing anything else when a new message is received
+	// SubjectTransformConfig is for applying a subject transform (to matching
+	// messages) before doing anything else when a new message is received.
 	SubjectTransformConfig struct {
-		Source      string `json:"src"`
+		// Source is the subject pattern to match incoming messages against.
+		Source string `json:"src"`
+
+		// Destination is the subject pattern to remap the subject to.
 		Destination string `json:"dest"`
 	}
 
-	// RePublish is for republishing messages once committed to a stream. The original
-	// subject is remapped from the subject pattern to the destination pattern.
+	// RePublish is for republishing messages once committed to a stream. The
+	// original subject is remapped from the subject pattern to the destination
+	// pattern.
 	RePublish struct {
-		Source      string `json:"src,omitempty"`
+		// Source is the subject pattern to match incoming messages against.
+		Source string `json:"src,omitempty"`
+
+		// Destination is the subject pattern to republish the subject to.
 		Destination string `json:"dest"`
-		HeadersOnly bool   `json:"headers_only,omitempty"`
+
+		// HeadersOnly is a flag to indicate that only the headers should be
+		// republished.
+		HeadersOnly bool `json:"headers_only,omitempty"`
 	}
 
 	// Placement is used to guide placement of streams in clustered JetStream.
 	Placement struct {
-		Cluster string   `json:"cluster"`
-		Tags    []string `json:"tags,omitempty"`
+		// Cluster is the name of the cluster to which the stream should be
+		// assigned.
+		Cluster string `json:"cluster"`
+
+		// Tags are used to match streams to servers in the cluster. A stream
+		// will be assigned to a server with a matching tag.
+		Tags []string `json:"tags,omitempty"`
 	}
 
 	// StreamSource dictates how streams can source from other streams.
 	StreamSource struct {
-		Name              string                   `json:"name"`
-		OptStartSeq       uint64                   `json:"opt_start_seq,omitempty"`
-		OptStartTime      *time.Time               `json:"opt_start_time,omitempty"`
-		FilterSubject     string                   `json:"filter_subject,omitempty"`
+		// Name is the name of the stream to source from.
+		Name string `json:"name"`
+
+		// OptStartSeq is the sequence number to start sourcing from.
+		OptStartSeq uint64 `json:"opt_start_seq,omitempty"`
+
+		// OptStartTime is the timestamp of messages to start sourcing from.
+		OptStartTime *time.Time `json:"opt_start_time,omitempty"`
+
+		// FilterSubject is the subject filter used to only replicate messages
+		// with matching subjects.
+		FilterSubject string `json:"filter_subject,omitempty"`
+
+		// SubjectTransforms is a list of subject transforms to apply to
+		// matching messages.
```

Comment:
worth pointing out that this is also a way to do multiple filter subjects

Reply:
Added:
```go
		// SubjectTransforms is a list of subject transforms to apply to
		// matching messages.
		//
		// Subject transforms on sources and mirrors are also used as subject
		// filters with optional transformations.
```